torch/_decomp/decompositions.py

```diff
@@ -1226,9 +1226,9 @@ def native_group_norm_backward(
     ds = torch.mul(grad_output, input).view(N, C, HxW).sum(dim=[2])
     db = grad_output.view(N, C, HxW).sum(dim=[2])
 
-    d_input: Optional[Tensor] = None
-    d_gamma: Optional[Tensor] = None
-    d_bias: Optional[Tensor] = None
+    d_input = None
```

Comment:
Why does removing these types help? The typing should be fixed, not removed.

Reply:
For this specific line, with `allow_redefinition=True`: https://github.com/pytorch/pytorch/actions/runs/4999748262/jobs/8956555243?pr=101528#step:10:1326


```python
d_input = (
    torch.mul(grad_output.reshape(N, group, cpg, HxW), c1)
    + torch.mul(input.reshape(N, group, cpg, HxW), c2)
    + c3
reveal_type(d_input)  # note: Revealed type is "Union[torch._tensor.Tensor, None]"
)
```
mypy thinks the result of this is somehow `Optional[Tensor]`.

With `allow_redefinition=False`, the revealed type is `"torch._tensor.Tensor"` as expected, so I suppose this is a bug with `allow_redefinition` stemming from how it is able to infer types from the above expression?

I would be open to using a `type: ignore[union-attr]` here instead of removing the type declarations.

Reply:
Actually I found the source of the error 1f810e0ac1b06a91b4f6fa11ca529d22c65d37f6 😵‍💫 really obscure